examples/run_glue_tpu.py

```diff
@@ -256,14 +257,29 @@ def evaluate(args, model, tokenizer, prefix="", disable_logging=False):
             preds = np.squeeze(preds)
         result = compute_metrics(eval_task, preds, out_label_ids)
         results.update(result)
+        results['eval_loss'] = eval_loss.item()
+
+        # Average all metrics from each shard
```

Comment:
what are some of the metrics? does it make sense to avg all of them? some metrics may be additive.

Reply:
f1, accuracy, eval_loss, acc_and_f1 (avg of the two)

I checked they all make sense averaged.

Reply:
Yep ignore that :D Not additive as discussed. Will update PR.

Reply:
I'd be surprised if `global f1 == np.mean(local f1s)`. That's probably not true, let's verify on paper what the true formula to get global f1 is.

Reply:
Updated to sync the pred/label tensors directly instead (these shouldn't be that big for finetuning tasks; single integer 0/1 per example). This way we don't have to have some custom aggregation per metric and don't touch upstream core code.